plugins/modules/aci_qos_custom_policy.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Gaspard Micol (@gmicol) <gmicol@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: aci_qos_custom_policy
+short_description: Manage QoS Custom Policy (qos:CustomPol)
+description:
+- Manage QoS Custom Policies for tenants on Cisco ACI fabrics.
+- The custom QoS policy enables different levels of service to be assigned to network traffic,
+  including specifications for the Differentiated Services Code Point (DSCP) value(s), and the 802.1p Dot1p priority.
+options:
+  tenant:
+    description:
+    - The name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  qos_custom_policy:
+    description:
+    - The name of the QoS Custom Policy.
+    type: str
+    aliases: [ qos_custom_policy_name, name ]
+  description:
+    description:
+    - The description for the QoS Custom Policy.
+    type: str
+    aliases: [ descr ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.aci.aci
+- cisco.aci.annotation
+- cisco.aci.owner
+
+notes:
+- The C(tenant) must exist before using this module in your playbook.
+  The M(cisco.aci.aci_tenant) can be used for this.
+seealso:
+- module: cisco.aci.aci_tenant
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC classes
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- Gaspard Micol (@gmicol)
+"""
+
+EXAMPLES = r"""
+- name: Add a new QoS Custom Policy
+  cisco.aci.aci_qos_custom_policy:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    qos_custom_policy: my_qos_custom_policy
+    state: present
+  delegate_to: localhost
+
+- name: Delete an QoS Custom Policy
+  cisco.aci.aci_qos_custom_policy:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    qos_custom_policy: my_qos_custom_policy
+    state: absent
+  delegate_to: localhost
+
+- name: Query a QoS Custom Policy
+  cisco.aci.aci_qos_custom_policy:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    qos_custom_policy: my_qos_custom_policy
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query all QoS Custom Policies in my_tenant
+  cisco.aci.aci_qos_custom_policy:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    state: query
+  delegate_to: localhost
+  register: query_result
```

Comment:
put delete at the end as @sajagana suggested 

```suggestion
- name: Query a QoS Custom Policy
  cisco.aci.aci_qos_custom_policy:
    host: apic
    username: admin
    password: SomeSecretPassword
    tenant: my_tenant
    qos_custom_policy: my_qos_custom_policy
    state: query
  delegate_to: localhost
  register: query_result

- name: Query all QoS Custom Policies in my_tenant
  cisco.aci.aci_qos_custom_policy:
    host: apic
    username: admin
    password: SomeSecretPassword
    tenant: my_tenant
    state: query
  delegate_to: localhost
  register: query_result

- name: Delete an QoS Custom Policy
  cisco.aci.aci_qos_custom_policy:
    host: apic
    username: admin
    password: SomeSecretPassword
    tenant: my_tenant
    qos_custom_policy: my_qos_custom_policy
    state: absent
  delegate_to: localhost
```